app/src/main/java/protect/card_locker/ManageGroupsActivity.java

```diff
@@ -228,20 +228,23 @@ public void onMoveUpButtonClicked(View view) {
     @Override
     public void onEditButtonClicked(View view) {
         Intent intent = new Intent(this, ManageGroupActivity.class);
-        intent.putExtra("group", getGroupName(view));
+        final String groupName = getGroupName(view);
+        Group group = DBHelper.getGroupByName(mDatabase, groupName);
+        intent.putExtra("group", group._id);
         startActivity(intent);
     }
 
     @Override
     public void onDeleteButtonClicked(View view) {
         final String groupName = getGroupName(view);
+        Group group = DBHelper.getGroupByName(mDatabase, groupName);
 
         AlertDialog.Builder builder = new MaterialAlertDialogBuilder(this);
         builder.setTitle(R.string.deleteConfirmationGroup);
         builder.setMessage(groupName);
 
         builder.setPositiveButton(getString(R.string.ok), (dialog, which) -> {
-            DBHelper.deleteGroup(mDatabase, groupName);
+            DBHelper.deleteGroup(mDatabase, group._id);
```

Comment:
Makes more sense to me if this uses the ID because you're not doing anything with the name and the ID should be the canonical way to refer to a group.

Reply:
But the id is being used here, or am I seeing it wrong?

Reply:
Ah I see what you mean, the id is only retrieved via the groupName. But I don't know how to handle this, as the only info that can be extracted from the view is the group name...